src/sentry/hybridcloud/tasks/deliver_webhooks.py

```diff
@@ -0,0 +1,213 @@
+    queue="webhook.control",
+    silo_mode=SiloMode.CONTROL,
+)
+def deliver_webhook(payload_id: int) -> None:
+    """
+    Deliver a single webhook. If delivery does not raise the attempt is considered a success
+    and the webhookpayload is deleted.
+    """
+    try:
+        payload = WebhookPayload.objects.get(id=payload_id)
+    except WebhookPayload.DoesNotExist:
+        logger.info("deliver_webhook.invalid_record", extra={"id": payload_id})
+        return
+
+    try:
+        perform_request(payload)
+
+        payload.delete()
+    except DeliveryFailed as err:
+        metrics.incr(
+            "hybridcloud.deliver_webhook.delivery_failed", tags={"attempts": payload.attempts}
+        )
+        logger.info(
+            "deliver_webhook.delivery_failed",
+            extra={
+                "error": str(err),
+                "payload_id": payload.id,
+                "attempt": payload.attempts,
+                "mailbox_name": payload.mailbox_name,
+            },
+        )
+
+
+class DeliveryFailed(Exception):
+    """
+    Used to signal an expected delivery failure.
+    """
+
+    pass
+
+
+def perform_request(payload: WebhookPayload) -> None:
+    logging_context: dict[str, str | int] = {
+        "payload_id": payload.id,
+        "attempt": payload.attempts,
+    }
+    region = get_region_by_name(name=payload.region_name)
+
+    try:
+        client = RegionSiloClient(region=region)
+        with metrics.timer(
+            "hybridcloud.deliver_webhooks.send_request",
+            tags={"destination_region": region.name},
+        ):
+            logging_context["region"] = region.name
+            logging_context["request_method"] = payload.request_method
+            logging_context["request_path"] = payload.request_path
+
+            headers = json.loads(payload.request_headers)
+            response = client.request(
+                method=payload.request_method,
+                path=payload.request_path,
+                headers=headers,
+                # We need to send the body as raw bytes to avoid interfering with webhook signatures
+                data=payload.request_body.encode("utf-8"),
+                json=False,
+            )
+        logger.info(
+            "webhook_proxy.complete",
+            extra={
+                "status": getattr(
+                    response, "status_code", 204
+                ),  # Request returns empty dict instead of a response object when the code is a 204
+                **logging_context,
+            },
+        )
+    except SiloClientError as err:
+        metrics.incr(
+            "hybridcloud.deliver_webhooks.failure",
+            tags={"reason": "silo_client_error", "destination_region": region.name},
+        )
+        logger.warning(
+            "hybridcloud.deliver_webhooks.failure",
+            extra=logging_context,
+        )
+        sentry_sdk.capture_exception(err)
+        raise DeliveryFailed() from err
```

Comment:
This exception triggers a re-attempt. I've not built out the logic to drop hooks after they have reached the max attempts yet.